pkg/collectsub/datasource/csubsource/csubsource.go

```diff
@@ -57,7 +57,6 @@ func (d *csubDataSources) GetDataSources(ctx context.Context) (*datasource.DataS
 		return nil, err
 	}
 	ds := entriesToSources(ctx, entries)
-	d.lastEntries = ds
```

Comment:
what is the reason for this change?

Reply:
This is what causes the data race (thanks to @lumjjb for finding it). To my understanding, this is an optimization, to write to `lastEntries` before returning. I think it's a small optimization, so I removed it instead of adding locking to the csubDataSources struct.

Reply:
For reference, the failing test can be seen in the unit tests of [this](https://github.com/guacsec/guac/actions/runs/6344018675) ci job (I force pushed, so the CI run isn't visible in this PR thread anymore).